validator/client/src/main/java/tech/pegasys/teku/validator/client/ValidatorTimingActions.java

```diff
@@ -97,8 +105,30 @@ public void onAttestationAggregationDue(final UInt64 slot) {
   }
 
   @Override
-  public void onAttesterSlashing(final AttesterSlashing attesterSlashing) {}
+  public void onAttesterSlashing(final AttesterSlashing attesterSlashing) {
+    delegates.forEach(delegates -> delegates.onAttesterSlashing(attesterSlashing));
+    final Set<UInt64> slashedIndices = attesterSlashing.getIntersectingValidatorIndices();
+    maybeValidatorSlashedAction.ifPresent(
+        validatorSlashingAction -> {
+          final List<BLSPublicKey> slashedPublicKeys =
+              slashedIndices.stream()
+                  .map(slashedIndex -> validatorIndexProvider.getPublicKey(slashedIndex.intValue()))
+                  .filter(Optional::isPresent)
+                  .map(Optional::get)
+                  .collect(Collectors.toList());
+          validatorSlashingAction.perform(slashedPublicKeys);
+        });
+  }
 
   @Override
-  public void onProposerSlashing(final ProposerSlashing proposerSlashing) {}
+  public void onProposerSlashing(final ProposerSlashing proposerSlashing) {
+    delegates.forEach(delegates -> delegates.onProposerSlashing(proposerSlashing));
+    final UInt64 slashedIndex = proposerSlashing.getHeader1().getMessage().getProposerIndex();
```

Comment:
Let's start with `maybeValidatorSlashedAction.ifPresent(` to avoid doing anything when it's not present

Reply:
Oh right! Updated